nautobot/users/views.py

```diff
@@ -250,17 +250,29 @@ class SavedViewUIViewSet(
     table_class = SavedViewTable
     action_buttons = ("export",)
 
+    def get_queryset(self):
+        """
+        Get the list of items for this view.
+        All users should be able to see saved views so we do not apply extra permissions.
+        """
+        return self.queryset
+
     def get_required_permission(self):
         """
         Obtain the permissions needed to perform certain actions on a model.
         """
         queryset = self.get_queryset()
         try:
             actions = [self.get_action()]
+            actions_copy = [self.get_action()]
             for i, item in enumerate(actions):
                 # Enforce users:change_savedview instead of users:clear_savedview
                 if item == "clear":
-                    actions[i] = "change"
+                    actions_copy[i] = "change"
+                # All users should be able to view existing saved views and add new saved views.
+                if item in ["add", "view"]:
+                    actions_copy.pop(i)
+            actions = actions_copy
```

Comment:
Not sure I understand what we're doing here or why `actions_copy` is needed. Mind adding some comments?